examples/platform/linux/AppMain.h

```diff
@@ -22,13 +22,17 @@
 #include <controller/CHIPDeviceController.h>
 #include <controller/CommissionerDiscoveryController.h>
 #include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
 #include <platform/CHIPDeviceLayer.h>
 #include <platform/PlatformManager.h>
 #include <transport/TransportMgr.h>
 
 #include "Options.h"
 
-int ChipLinuxAppInit(int argc, char * const argv[], chip::ArgParser::OptionSet * customOptions = nullptr);
+// Applications can optionally provide the endpoint id of a secondary network
+// commissioning endpoint, if one is supported.
+int ChipLinuxAppInit(int argc, char * const argv[], chip::ArgParser::OptionSet * customOptions = nullptr,
+                     const chip::EndpointId * secondaryNetworkCommissioningEndpoit = nullptr);
```

Comment:
how about using Optional<> instead of null for missing.
Maybe in a future PR we can start using argument builders as the number of arguments to this starts increasing and we seem to want defaults.

Reply:
> how about using Optional<> instead of null for missing.

Can do.